satellite/console/service.go

```diff
@@ -165,6 +165,133 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 	return u, nil
 }
 
+func (s *Service) AddNewPaymentMethod(ctx context.Context, paymentMethodToken string, isDefault bool, projectID uuid.UUID, userID uuid.UUID) (payment ProjectPayment, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	info, err := s.store.UserPayments().Get(ctx, userID)
+	if err != nil {
+		return ProjectPayment{}, err
+	}
+
+	params := payments.AddPaymentMethodParams{
+		Token:      paymentMethodToken,
+		CustomerID: string(info.CustomerID),
+	}
+
+	method, err := s.pm.AddPaymentMethod(ctx, params)
+	if err != nil {
+		return ProjectPayment{}, err
+	}
+
+	projectPaymentInfo := ProjectPayment{ProjectID: projectID,
+		PayerID:         userID,
+		PaymentMethodID: []byte(method.ID),
+		CreatedAt:       time.Now(),
+		IsDefault:       isDefault}
+
+	create, err := s.store.ProjectPayments().Create(ctx, projectPaymentInfo)
+	if err != nil {
+		return ProjectPayment{}, err
+	}
+
+	return *create, nil
+}
+
+func (s *Service) SetDefaultPaymentMethod(ctx context.Context, projectPaymentID uuid.UUID, projectID uuid.UUID) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	_, err = GetAuth(ctx)
+	if err != nil {
+		return err
+	}
+
+	projectPayment, err := s.store.ProjectPayments().GetDefaultByProjectID(ctx, projectID)
+	if err != nil {
+		return err
+	}
+	projectPayment.IsDefault = false
+
+	err = s.store.ProjectPayments().Update(ctx, *projectPayment)
+	if err != nil {
+		return err
+	}
+
+	projectPayment, err = s.store.ProjectPayments().GetByID(ctx, projectPaymentID)
+	if err != nil {
+		return err
+	}
+	projectPayment.IsDefault = true
+
+	err = s.store.ProjectPayments().Update(ctx, *projectPayment)
+
+	return err
+}
+
+func (s *Service) DeletePaymentMethod(ctx context.Context, projectPayment uuid.UUID) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	_, err = GetAuth(ctx)
+	if err != nil {
+		return err
+	}
+
+	return s.store.ProjectPayments().Delete(ctx, projectPayment)
+}
+
+// GetProjectPaymentMethods retrieves project payment methods
+func (s *Service) GetProjectPaymentMethods(ctx context.Context, projectID uuid.UUID) ([]ProjectPayment, error) {
```

Comment:
we should return `payments` type

`project_payments` supposed to be just a mapping and implementation detail